Fixes menu collapsing to ellipse

douglance opened the pull request:
Closes #185

douglance commented:
![Screen Shot 2021-12-27 at 2 04 59 PM](https://user-images.githubusercontent.com/4741454/147500206-4faadb39-55e2-4bfb-a04a-0803117d5f0d.png)

alex-miller-0 commented:
@douglance is there any way to actually collapse the menu on mobile? It would be nice to stretch the page across the whole width of a small device

douglance commented:
@alex-miller-0 Like this?

https://user-images.githubusercontent.com/4741454/147590349-d99ba9cf-8206-48de-91fc-ded0bb21baf2.mp4

alex-miller-0 commented:
@douglance Yes! Looks good. Can you work on formatting for those screens as well? Looks like the redesign messed up some things like this: 
![image](https://user-images.githubusercontent.com/7378490/147592506-11087dac-d6c0-4d38-8adc-d61d963197ee.png)

douglance commented:
I pushed that hamburger menu change. 

I also fixed those floating tooltips, the margins on the Landing, and also fixed the giant lattice image on the Landing that was bothering me... 😄

alex-miller-0 commented on src/components/main.js:
```diff
@@ -267,8 +269,10 @@ class Main extends React.Component {
-    this.setState({ menuItem: key })
+  handleMenuChange ({ key }) {
+    const stateUpdate = { menuItem: key }
+    if (this.isMobile()) stateUpdate.collapsed = true
```
What is the purpose of this line? You've already set `collasped: true` on L74

douglance replied:
When the user taps on a menu item, this line collapses the menu for them. 

L74 collapses the menu for mobile uses when the Main component is mounted, so because it doesn't unmount when the content changes, L74 doesn't fire again.

alex-miller-0 commented on src/components/main.js:
```diff
@@ -70,6 +71,7 @@ class Main extends React.Component {
     // Listen for window resize
     window.addEventListener('resize', this.updateWidth);
 
+    if (this.isMobile()) this.setState({collapsed: true})
```
You should put a handler in `this.updateWidth` as well. I noticed that if I shrink to mobile size, collapse the menu to the hamburger, and then resize to full desktop, the menu disappears completely. Think you could add something like the following to `this.updateWidth`:

```
if (this.isMobile() && !this.state.collapsed) {
  this.setState({ collapsed: true })
} else if (!this.isMobile() && this.state.collapsed) {
  this.setState({ collapsed: false })
}

douglance replied:
That worked well without modification. 👍

alex-miller-0 reviewed (commented):
Smol comments. Looks mostly good. Nice resizing of the massive image :)

alex-miller-0 reviewed (approved):
🍻
